pysph/sph/wc/linalg.py

```diff
@@ -21,6 +21,56 @@ def dot(a=[0.0, 0.0], b=[0.0, 0.0], n=3):
     return result
 
 
+def deteminant_2d(A=[0.0, 0.0]):
+    return A[0]*A[3] - A[1]*A[2]
+
+
+def deteminant_3d(A=[0.0, 0.0]):
+    c1 = deteminant_2d([A[4], A[5], A[7], A[8]])
+    c2 = deteminant_2d([A[3], A[5], A[6], A[8]])
+    c3 = deteminant_2d([A[3], A[4], A[6], A[7]])
+    result = A[0] * c1 -  A[1] * c2 + A[2] * c3
+    return result
+
+
+def deteminant_4d(A=[0.0, 0.0]):
+    c1 = deteminant_3d([A[5], A[6], A[7],
+                   A[9], A[10], A[11],
+                   A[13], A[14], A[15]])
+    c2 = deteminant_3d([A[4], A[6], A[7],
+                   A[8], A[10], A[11],
+                   A[12], A[14], A[15]])
+    c3 = deteminant_3d([A[4], A[5], A[7],
+                   A[8], A[9], A[11],
+                   A[12], A[13], A[15]])
+    c4 = deteminant_3d([A[4], A[5], A[6],
+                   A[8], A[9], A[10],
+                   A[12], A[13], A[14]])
+    result = A[0] * c1 - A[1] * c2 + A[2] * c3 - A[3] * c4
+    return result
+
+
+def replace_vector_in_matrix(A=[0.0, 0.0], b=[0.0, 0.0],
+                             pos=2, dim=3, result=[0.0, 0.0]):
+    i, index, next = declare('int', 3)
+    if pos + 1 <= dim:
+        index = 0
+        next = 0
+        for i in range(dim*dim):
+            if index == pos:
+                result[i] = b[next]
+                next += 1
+            else:
+                result[i] = A[i]
+
+            index += 1
+            if index == dim:
+                index = 0
```

Comment:
By doing this you can save on the additional condition.
```
for i in range(dim):
    for j in range(dim):
        if j = pos:
            result[i*dim + j] = b[i]
        else:
            result[i*dim + j] = A[i*dim + j]
```

Reply:
done, Thanks.